src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -50,13 +49,19 @@ internal class FoldingRangeEndpoint : IFoldingRangeHandler
             _logger = loggerFactory.CreateLogger<FoldingRangeEndpoint>();
         }
 
-        public FoldingRangeRegistrationOptions GetRegistrationOptions(FoldingRangeCapability capability, ClientCapabilities clientCapabilities)
-            => new()
+        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
+        {
+            const string AssociatedServerCapability = "foldingRangeProvider";
+
+            var registrationOptions = new FoldingRangeOptions()
             {
-                DocumentSelector = RazorDefaults.Selector,
+                WorkDoneProgress = false,
```

Comment:
can omit the `WorkDoneProgress` bit and just make this `var registrationOptions = new FoldingRangeOptions();`